src/frontend/js/components/PurchaseButton/index.spec.tsx

```diff
@@ -102,11 +103,14 @@ describe('PurchaseButton', () => {
   it('shows cta to open sale tunnel when user is authenticated', async () => {
     const courseCode = '00000';
     const product = ProductFactory().one();
+    const orderQueryParameters = {
+      course_code: courseCode,
+      product_id: product.id,
+      state: [...ACTIVE_ORDER_STATES, ...PURCHASABLE_ORDER_STATES],
```

Comment:
As `[...ACTIVE_ORDER_STATES, ...PURCHASABLE_ORDER_STATES]` is used many times, what about adding `NON_CANCELED_ORDER_STATES` in [types](https://github.com/openfun/richie/blob/65bb019557cae6ff4d7548e9186b905e888e6076/src/frontend/js/types/Joanie.ts#L296) ?